### PR TITLE
feat(MRS): support to update the cluster name

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
Running tool: /usr/local/go/bin/go test -timeout 9000000000s -run ^TestAccMrsMapReduceCluster_basic$ github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/mrs -v

=== RUN   TestAccMrsMapReduceCluster_basic
=== PAUSE TestAccMrsMapReduceCluster_basic
=== CONT  TestAccMrsMapReduceCluster_basic
--- PASS: TestAccMrsMapReduceCluster_basic (722.74s)
PASS
ok  	github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/mrs	722.774s
```

### CODE REVIEW

File: huaweicloud/services/mrs/resource_huaweicloud_mapreduce_cluster.go [hunk 1]
```diff
@@ -1007,13 +1006,62 @@ func updateMRSClusterNodes(ctx context.Context, d *schema.ResourceData, client *
...
+func updateClusterName(d *schema.ResourceData, cfg *config.Config) error {
+	region := cfg.GetRegion(d)
+
+	var (
+		updateNameHttpUrl = "v2/{project_id}/clusters/{id}/cluster-name"
+		updateNameProduct = "mrs"
+	)
+	updateNameClient, err := cfg.NewServiceClient(updateNameProduct, region)
+	if err != nil {
+		return fmt.Errorf("error creating DWS Client: %s", err)
```
Comment (ShiChangkuo) [inline huaweicloud/services/mrs/resource_huaweicloud_mapreduce_cluster.go]:
```suggestion
		return fmt.Errorf("error creating MRS client: %s", err)
```

File: huaweicloud/services/mrs/resource_huaweicloud_mapreduce_cluster.go [hunk 2]
```diff
@@ -1007,13 +1006,62 @@ func updateMRSClusterNodes(ctx context.Context, d *schema.ResourceData, client *
...
+	if err != nil {
+		return fmt.Errorf("error updating name of MRS cluster: %s", err)
+	}
+
+	return nil
+}
+
+func buildUpdateNameBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"cluster_name": utils.ValueIngoreEmpty(d.Get("name")),
```
Comment (ShiChangkuo) [inline huaweicloud/services/mrs/resource_huaweicloud_mapreduce_cluster.go]: `name` is required, so I think IngoreEmpty is unnecessary.

File: huaweicloud/services/mrs/resource_huaweicloud_mapreduce_cluster.go (hunk 1 above)
Comment (hmbig2) [reply huaweicloud/services/mrs/resource_huaweicloud_mapreduce_cluster.go]: fixed

File: huaweicloud/services/mrs/resource_huaweicloud_mapreduce_cluster.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/services/mrs/resource_huaweicloud_mapreduce_cluster.go]: done

**Pull request merged**